### PR TITLE
[Fix-4721][worker]The shell background starts the YARN task scenario, and the kill function is abnormal

### LINKED ISSUE #4721: [Bug][worker] The shell background starts the YARN task scenario, and the kill function is abnormal
**Describe the bug**
The shell background starts the YARN task scenario, and the kill function is abnormal

**To Reproduce**
1.Create a new shell task to start the yarn task in the background    
2.While the yarn task is running, the page clicks on this task kill function

**Expected behavior**

**Screenshots**

**Which version of Dolphin Scheduler:**
 -[1.3.4-release,dev]

---

**描述 bug**
在shell后台启动yarn任务场景，kill功能异常

**To Reproduce**
1.新建一个shell任务，用于后台启动yarn任务
2.在yarn任务正在运行时，页面点击此任务kill功能

异常：
页面点击kill后，工作流实例显示状态为准备停止

image

image

worker日志

任务正在运行日志
image

任务kill时日志

image

### PR DESCRIPTION
**Pull request by zhuangchong**
## What is the purpose of the pull request
this closes #4721 
fix the shell background starts the YARN task scenario, and the kill function is abnormal

## Verify this pull request
This change added tests and can be verified as follows:
  - *Manually verified the change by testing locally.*

### CODE REVIEW

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/StringUtils.java [hunk 1]
```diff
@@ -69,4 +72,40 @@ public static String defaultIfBlank(String str, String defaultStr) {
     public static boolean equalsIgnoreCase(String str1, String str2) {
         return str1 == null ? str2 == null : str1.equalsIgnoreCase(str2);
     }
+
```
Comment (CalvinKirs) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/StringUtils.java:134]: Please add comments for public methods.

**Review by CalvinKirs — CHANGES_REQUESTED** [review]

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/StringUtils.java (hunk 1 above)
Comment (zhuangchong) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/StringUtils.java:134]: done.

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Comment by CalvinKirs** [comment]
please resolve conflicts.thx

**Pull request merged**